include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -7082,7 +7082,7 @@ namespace VectorTools
         double my_values[3] = { mean_double.real(), mean_double.imag(), area };
         double global_values[3];
 
-        MPI_Allreduce (&my_values, &global_values, 3, MPI_DOUBLE,
+        MPI_Allreduce (my_values, global_values, 3, MPI_DOUBLE,
```

Comment:
The type of `&my_values` is `double (*)[3]`, i.e., pointer to array. Since `MPI_Allreduce` takes a `void*` as argument, the conversion works just fine.

The type of `my_values` is `double*`, which also converts correctly to `void*`. So I think both forms are equally valid.